google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java

```diff
@@ -232,79 +224,19 @@ protected void resetEditorFrom(@NotNull AppEngineDeploymentConfiguration configu
   @Override
   protected void applyEditorTo(@NotNull AppEngineDeploymentConfiguration configuration)
       throws ConfigurationException {
-    validateConfiguration();
-
     commonConfig.applyEditorTo(configuration);
+    commonConfig.setDeploymentProjectAndVersion(deploymentSource);
 
-    configuration.setModuleName(
-        ((AppEngineFlexibleFacet) appYamlCombobox.getSelectedItem()).getModule().getName());
-    CredentialedUser user = commonConfig.getProjectSelector().getSelectedUser();
-    if (user != null) {
-      configuration.setGoogleUsername(user.getEmail());
+    if (appYamlCombobox.getSelectedItem() != null) {
+      configuration.setModuleName(
+          ((AppEngineFlexibleFacet) appYamlCombobox.getSelectedItem()).getModule().getName());
     }
-    String environment = "";
-    if (deploymentSource instanceof AppEngineDeployable) {
-      environment = ((AppEngineDeployable) deploymentSource).getEnvironment().name();
-    }
-    configuration.setEnvironment(environment);
-    configuration.setUserSpecifiedArtifact(
-        deploymentSource instanceof UserSpecifiedPathDeploymentSource);
+
+    configuration.setEnvironment(AppEngineEnvironment.APP_ENGINE_FLEX);
     configuration.setUserSpecifiedArtifactPath(archiveSelector.getText());
+    configuration.setAppYamlPath(getAppYamlPath());
```

Comment:
similar to above comment - I suspect this (and Dockerfile) isn't needed. That is what the moduleName is for.